usage/rest-server/src/main/java/brooklyn/rest/resources/CatalogResource.java

```diff
@@ -454,8 +455,20 @@ private Response getCatalogItemIcon(CatalogItem<?, ?> result) {
     @SuppressWarnings("unchecked")
     private static <T> List<T> castList(List<? super T> list, Class<T> elementType) {
         List<T> result = Lists.newArrayList();
-        for (Object element : list) {
-            result.add((T) element);
+        Iterator<? super T> li = list.iterator();
+        while (li.hasNext()) {
+            try {
+                result.add((T) li.next());
+            } catch (Throwable throwable) {
+                if (throwable instanceof NoClassDefFoundError) {
```

Comment:
How is this possible? The destination class is already loaded (passed as argument).